tests/test_agents.py

```diff
@@ -3,24 +3,25 @@
 import itertools
 import json
 import re
+from pathlib import Path
 from typing import Any, cast
 from unittest.mock import patch
 
 import pytest
-from langchain.agents.openai_functions_agent.base import OpenAIFunctionsAgent
```

Comment:
The saved settings objects in /config still have OpenAIFunctionsAgent could we replace there with whichever one is closest.

Reply:
Let's use `ToolSelector` mainly so we don't require `ldp`

Reply:
Done